pandas/tests/test_strings.py

```diff
@@ -377,11 +377,36 @@ def test_str_cat_align_mixed_inputs(self, join):
         with tm.assert_raises_regex(ValueError, rgx):
             s.str.cat([t, z], join=join)
 
-    def test_str_cat_raises(self):
+    @pytest.mark.parametrize('box', [Series, Index])
+    def test_str_cat_raises(self, box):
         # non-strings hiding behind object dtype
-        s = Series([1, 2, 3, 4], dtype='object')
-        with tm.assert_raises_regex(TypeError, "unsupported operand type.*"):
-            s.str.cat(s)
+        s = box([1, 2, 3, 4], dtype='object')
+        message = 'Can only use .str accessor with string values'
+        with tm.assert_raises_regex(AttributeError, message):
+            s.str
+
+    @pytest.mark.parametrize('box', [Series, Index])
+    @pytest.mark.parametrize('other', [Series, Index])
+    def test_str_cat_all_na(self, box, other):
+        # check that all NaNs in caller / target work
+        s = Index(['a', 'b', 'c', 'd'])
```

Comment:
can you make this more transparent and don't use abbreviations. This is very hard to read

Reply:
Not sure I understand you here - this uses the same format as the other tests for `str.cat`, and I can't see any abbreviations.